explainer.md

```diff
@@ -18,96 +18,107 @@ permanent "block" policies, or
-   to miss since it's on the opposite side of the page._
-
-1. **Regret**: When users decline to provide a capability to a given site, it
-   seems quite reasonable for user agents to suppress that site's future
-   requests for the same capability in order to avoid annoyance or abuse. That
-   said, users change their minds for good reasons, but sometimes struggle to
-   understand how to express that new decision to the user agent (see example
-   3). In these cases, the user agent's desire to protect the user backfires,
-   leaving the user confused about how to make the site they want to use work
-   the way they want it to.User agents could make better decisions and provide
-   better re-prompt UI if they could make well-founded assumptions about the
-   nature of the user's interaction in the content area, and the user's intent.
-
-   ![](images/image3.png) \
-   _Example 3. An example where the user previously blocked camera and microphone
-   access, but has now just expressed a strong intention to re-enable them by clicking
-   the unmute buttons. Because the user agent has no insight into this interaction
-   in the content area, it is compelled to respect the user's previous decision.
-   Especially in a stressful scenario such as an important presentation, users will
-   struggle to navigate the settings surfaces to change the permission decision._
-
-Overall, attempts to optimize the trade-off between usability and interruptions
-have hit practical limits because, fundamentally, user agents
+There are four main challenges with the status quo:
+
+1.  **Insufficiency of existing mitigations** The present day permissions spam
+    and abuse mitigation approach has an architectural upper bound on user
+    protection because the model relies on the website to choose when to trigger
+    the permission request prompt rather than capturing a reliable signal of
+    user intent. Requiring a user gesture for the Permission API (or similar)
+    does not solve this problem as there are many ways of tricking a user into
+    providing such a gesture.
+
+1.  **Context**: Ideally, a site's developer will request access as part of a
+    contextual flow that helps users understand what's being asked for and why,
+    enabling quick and confident responses. Often, however, permission requests
+    are correlated poorly with user expectations, up to and including prompts
+    that can come out of nowhere (see example 1). This places a burden on user
+    agents' presentation of the request. The user agent has no semantic
+    understanding of events taking place in the content area prior to the
+    permission request. User agents could make better decisions and provide
+    better prompts if they could make well-founded assumptions about the nature
+    of the user's interaction in the content area, and the user's intent.
+
+    ![](images/image1.png) \
+    *Example 1. A notification permission prompt on a news site (contents
```

Comment:
This example seems contrived... for Notifications, the API doesn't require a user gesture to request permissions. That's a flaw in the Notifications API (which we should fix collectively as user agents), but the notification prompt is not resulting from the user clicking - the site can show the permission prompt at any time.

Reply:
This is a real-world scenario that I've personally experienced several times. The notifications API might not require a user gesture, but for some reason it seems that some sites will request the notification permission as soon as a click happens in the page.